src/entry.c

```diff
@@ -690,6 +694,10 @@ vstumpless_new_entry( int facility,
   }
 
   effective_msgid = msgid ? msgid : "-";
+  if( !validate_msgid_length( effective_msgid ) ) {
+      goto rejected_msgid;
```

Comment:
Use the existing `fail_msgid` label here instead of adding a new one. As is, there is a memory leak where the `entry` and `app_name` will be allocated, but not freed upon an error condition. The existing failure branch already handles this case for you.